src/stores/actor.ts

```diff
@@ -0,0 +1,146 @@
+/**
+ * SPDX-FileCopyrightText: 2019 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+/**
+ * This store helps to identify a the current actor in all cases.
+ * In Talk not every user is a local nextcloud user, so identifying
+ * solely by userId is not enough.
+ * If an as no userId, they are a guest and identified by actorType + sessionId.
+ */
+
+import type { Participant } from '../types/index.ts'
+
+import { loadState } from '@nextcloud/initial-state'
+import { defineStore } from 'pinia'
+import { computed, ref } from 'vue'
+import { ATTENDEE, PARTICIPANT } from '../constants.ts'
+import { getTeams } from '../services/teamsService.ts'
+
+interface NextcloudUser {
+	uid: string
+	displayName: string | null
+}
+
+export const useActorStore = defineStore('actor', () => {
+	const userId = ref<string | null>(null)
+	const sessionId = ref<string | null>(null)
+	const attendeeId = ref<number | null>(null)
+	const actorId = ref<string | null>(null)
+	const actorType = ref<string | null>(null)
+	const displayName = ref<string>('')
+	const actorGroups = ref<string[]>(loadState('spreed', 'user_group_ids', []))
+	const actorTeams = ref<string[]>([])
+
+	const isActorUser = computed(() => actorType.value === ATTENDEE.ACTOR_TYPE.USERS)
+	const isActorGuest = computed(() => actorType.value === ATTENDEE.ACTOR_TYPE.GUESTS)
+	const getParticipantIdentifier = computed(() => ({
```

Comment:
As this is a value, not a function
```suggestion
	const participantIdentifier = computed(() => ({
```